xapi/tablecodec/tablecodec.go

```diff
@@ -138,6 +138,9 @@ func unflatten(datum types.Datum, ft *types.FieldType) (types.Datum, error) {
 		datum.SetValue(dur)
 		return datum, nil
 	case mysql.TypeNewDecimal:
+		if datum.Kind() == types.KindMysqlDecimal {
```

Comment:
When the decimal datum is decoded from index, it's kind is already KindMysqlDecimal, when the decimal datum is decoded from column value, it's kind is String, so we need to parse it.